### DISCUSSION TITLE
require a .bin with VL53L0x included

### DISCUSSION THREAD

**Discussion by lwyl5744** [arendst/Tasmota#19366] [Support request]
I need a bin file that allows the use of a VL53L0x radar module

**Comment by sfromis** [comment]
Assuming ESP8266, upgrade to `tasmota-sensors.bin.gz`

**Reply by lwyl5744** [reply]
Thanks but that doesnt seem to work either.
I can use the device on an arduino so I know it works.

**Reply by barbudor** [reply]
As per the documentation https://tasmota.github.io/docs/BUILDS/#available-features-and-sensors
Driver for VL53L0x is included in the 'tasmota-sensors'
If you can't see it working, double check the docs:
- is the device found at address 0x29 with a i2cscan command
- is the driver 31 enabled with command i2cdriver

Some more information in the dedicated doc https://tasmota.github.io/docs/VL53Lxx/

**Reply by barbudor** [reply]
Preferably have only driver 31 enabled and no there's as i2c drivers cannot identify the specific device at a given address
Some other drivers are looking for devices at 0x29 (like TSL light sensors) and should be disabled

**Reply by sfromis** [reply]
From a running Tasmota device, I just confirmed that the VL53L0X driver (but not VL53L1X) is included in tasmota-sensors. Assuming regular precompiled build (current versions), the TSL light sensors are not included in the build.

**Comment by stefanbode** [comment]
If the team has no problem I can share my TASMOCompiler and you can create your own Firmware: http://elocm.synology.me:3000
for the VL53L0X you must add in the last section some additional defines.
```
//  #define USE_VL53L0X                            // [I2cDriver31] Enable VL53L0x time of flight sensor (I2C address 0x29) (+4k code)
//    #define VL53L0X_XSHUT_ADDRESS 0x78           //   VL53L0X base address when used with XSHUT control
//  #define USE_VL53L1X                            // [I2cDriver54] Enable VL53L1X time of flight sensor (I2C address 0x29) using Pololu VL53L1X library (+2k9 code)
//    #define VL53L1X_XSHUT_ADDRESS 0x78           //   VL53L1X base address when used with XSHUT control
//    #define VL53L1X_DISTANCE_MODE Long           //   VL53L1X distance mode : Long | Medium | Short
```
More info here: https://tasmota.github.io/docs/VL53Lxx/

**Reply by sfromis** [reply]
VL53L0X support is already included in tasmota-sensors. No need for a build to use this, unless it is for other reasons, or to also set `VL53L0X_XSHUT_ADDRESS`.

Of course, independently of this, another TasmoCompiler instance might be useful to some :-)

**Reply by lwyl5744** [reply]
Thanks for all the information.
I am running the latest tasmota with sensors on an esp 8266.
I once got a message from a I2CScan that the address 29 was available.
Tried the information as per the documents in particular the backlog that turns off some drivers and turns on driver 31 all to no avail.
I am at a loss.

**Reply by stefanbode** [reply]
Please post the result of I2cscan

**Reply by sfromis** [reply]
And of `I2CDriver`

**Reply by lwyl5744** [reply]
I dont understand the question, but I can only use what Tasmota has in the Sensors.bin

**Reply by sfromis** [reply]
Well, I thought that the point was trying to get your VL53L0X working with Tasmota, using the `tasmota-sensors` binary. And that sensor **is** included in that binary (as long as it is not a VL53L1X instead). And to make sure of your setup, post the output of those commands.
You can issue both at the same time using the command `Backlog I2CScan; I2CDriver`.

**Reply by lwyl5744** [reply]
I cant tell you what you request because as soon as I configure the esp8266  GPIO4 with SDA and GPIO5 with SCL with the VL53L0X  pin connections double checked and connected then hit SAVE the esp8266 reboots and the configuration is lost.
This does not happen with any other I2C device. Tried many times on different  ESP8266 boards.

**Reply by barbudor** [reply]
If it reboots and clear the config, then there is a crash at boot and the configuration is automatically erased as a failsafe

The only way to get proper logs thennis from serial
You need a serial terminal with the capability to log to file and share the from the point you configure sda/SCL until it reboots with configuration erased
